handlers/handlers.go

```diff
@@ -0,0 +1,368 @@
+	json.NewEncoder(w).Encode(response)
+}
+
+func GetTasksHandler(w http.ResponseWriter, r *http.Request, database *sql.DB) {
+	var tasks []models.Task
+
+	err := getTasks(database, &tasks)
+	if err != nil {
+		logAndReturnError(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	if tasks == nil {
+		tasks = []models.Task{}
+	}
+
+	resp := map[string]interface{}{
+		"tasks": tasks,
+	}
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(resp)
+}
+func UpdateTaskHandler(w http.ResponseWriter, r *http.Request, database *sql.DB) {
+	var task models.Task
+
+	if err := json.NewDecoder(r.Body).Decode(&task); err != nil {
+		http.Error(w, `{"error":"Ошибка десериализации JSON"}`, http.StatusBadRequest)
+		return
+	}
+
+	if task.Id == "" {
+		http.Error(w, `{"error":"Не указан идентификатор задачи"}`, http.StatusBadRequest)
+		return
+	}
+
+	if task.Title == "" {
+		http.Error(w, `{"error":"Не указан заголовок задачи"}`, http.StatusBadRequest)
+		return
+	}
+
+	if task.Date == "" {
+		task.Date = time.Now().Format("20060102")
+	}
+
+	if err := validateDateFormat(task.Date); err != nil {
+		http.Error(w, `{"error":"Неверный формат даты"}`, http.StatusBadRequest)
+		return
+	}
+
+	taskDate, err := time.Parse("20060102", task.Date)
+	if err != nil {
+		http.Error(w, `{"error":"Неверная дата"}`, http.StatusBadRequest)
+		return
+	}
+
+	now := time.Now()
+	if taskDate.Before(now) {
+		if task.Repeat != "" {
+			nextDate, err := services.NextDate(now, task.Date, task.Repeat)
+			if err != nil {
+				http.Error(w, `{"error":"Ошибка при вычислении следующей даты"}`, http.StatusBadRequest)
+				return
+			}
+			task.Date = nextDate
+		} else {
+			task.Date = now.Format("20060102")
+		}
+	}
+
+	query := "UPDATE scheduler SET date = ?, title = ?, comment = ?, repeat = ? WHERE id = ?"
+	res, err := database.Exec(query, task.Date, task.Title, task.Comment, task.Repeat, task.Id)
+	if err != nil {
+		http.Error(w, `{"error":"Ошибка обновления в базе данных"}`, http.StatusInternalServerError)
+		return
+	}
+
+	rowsAffected, err := res.RowsAffected()
+	if err != nil {
+		http.Error(w, `{"error":"Ошибка при получении результата обновления"}`, http.StatusInternalServerError)
+		return
+	}
+
+	if rowsAffected == 0 {
+		http.Error(w, `{"error":"Задача не найдена"}`, http.StatusNotFound)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.Write([]byte(`{}`))
+}
+
+func DoneHandler(w http.ResponseWriter, r *http.Request, database *sql.DB) {
+
+	taskID := r.URL.Query().Get("id")
+	if taskID == "" {
+		http.Error(w, `{"error":"Не указан идентификатор задачи"}`, http.StatusBadRequest)
+		return
+	}
+
+	var task models.Task
+
+	query := "SELECT id, date, title, comment, repeat FROM scheduler WHERE id = ?"
+
+	err := database.QueryRow(query, taskID).Scan(&task.Id, &task.Date, &task.Title, &task.Comment, &task.Repeat)
+	if err == sql.ErrNoRows {
```

Comment:
Ошибки лучше сравнивать, используя специальные функции 
```suggestion
	if errors.Is(err,sql.ErrNoRows) {
```